components/datadog/apps/mutatedbyadmissioncontroller/k8s.go

```diff
@@ -0,0 +1,87 @@
+package mutatedbyadmissioncontroller
+
+import (
+	"github.com/DataDog/test-infra-definitions/common/config"
+
+	"github.com/pulumi/pulumi-kubernetes/sdk/v3/go/kubernetes"
+	appsv1 "github.com/pulumi/pulumi-kubernetes/sdk/v3/go/kubernetes/apps/v1"
+	corev1 "github.com/pulumi/pulumi-kubernetes/sdk/v3/go/kubernetes/core/v1"
+	metav1 "github.com/pulumi/pulumi-kubernetes/sdk/v3/go/kubernetes/meta/v1"
+	"github.com/pulumi/pulumi/sdk/v3/go/pulumi"
+)
+
+type K8sComponent struct {
+	pulumi.ResourceState
+}
+
+func K8sAppDefinition(e config.CommonEnvironment, kubeProvider *kubernetes.Provider, namespace string, opts ...pulumi.ResourceOption) (*K8sComponent, error) {
+	opts = append(opts, pulumi.Provider(kubeProvider), pulumi.Parent(kubeProvider), pulumi.DeletedWith(kubeProvider))
+
+	k8sComponent := &K8sComponent{}
+	if err := e.Ctx.RegisterComponentResource("dd:apps", "mutated", k8sComponent, opts...); err != nil {
+		return nil, err
+	}
+
+	opts = append(opts, pulumi.Parent(k8sComponent))
+
+	ns, err := corev1.NewNamespace(e.Ctx, namespace, &corev1.NamespaceArgs{
+		Metadata: &metav1.ObjectMetaArgs{
+			Name: pulumi.String(namespace),
+		},
+	}, opts...)
+	if err != nil {
+		return nil, err
+	}
+
+	opts = append(opts, pulumi.Parent(ns))
+
+	if _, err := appsv1.NewDeployment(e.Ctx, "mutated", &appsv1.DeploymentArgs{
+		Metadata: &metav1.ObjectMetaArgs{
+			Name:      pulumi.String("mutated"),
+			Namespace: pulumi.String(namespace),
+			Labels:    pulumi.StringMap{"app": pulumi.String("mutated")},
+		},
+		Spec: &appsv1.DeploymentSpecArgs{
+			Replicas: pulumi.Int(1),
+			Selector: &metav1.LabelSelectorArgs{
+				MatchLabels: pulumi.StringMap{"app": pulumi.String("mutated")},
+			},
+			Template: &corev1.PodTemplateSpecArgs{
+				Metadata: &metav1.ObjectMetaArgs{
+					Labels: pulumi.StringMap{
+						"app":                             pulumi.String("mutated"),
+						"admission.datadoghq.com/enabled": pulumi.String("true"),
+						"tags.datadoghq.com/env":          pulumi.String("e2e"),
+						"tags.datadoghq.com/service":      pulumi.String("mutated"),
+						"tags.datadoghq.com/version":      pulumi.String("v0.0.1"),
+					},
+				},
+				Spec: &corev1.PodSpecArgs{
+					Containers: corev1.ContainerArray{
+						corev1.ContainerArgs{
+							Name:  pulumi.String("mutated"),
+							Image: pulumi.String("busybox"),
```

Comment:
Should we have a dummy image build as package as we did for redis, nginx and co?

Reply:
The thing is that I don’t really care about what image is running and what it is doing as [the corresponding test](https://github.com/DataDog/datadog-agent/pull/21653/files#diff-2bca1254c33d752a13e15a261a21b1282ec394f1e06fba2b8d850c4215cf5102R724-R739) can only validate that the pod spec has been mutated as expected.

Reply:
I'm thinking DockerHub rate limit that's why but yeah I imagine the image does not matter

Reply:
I’ve created an image for GHCR: https://github.com/DataDog/test-infra-definitions/pull/522/commits/d07477af6942be90c943a30cd30457c64a0f47c6